src/tests/team.rs

```diff
@@ -0,0 +1,234 @@
+
+    let body = r#"{"owners":["github:rust-lang:lang"]}"#;
+    let json = bad_resp!(middle.call(req.with_path("/api/v1/crates/foo/owners")
+                           .with_method(Method::Put)
+                           .with_body(body.as_bytes())));
+    assert!(json.errors[0].detail.contains("only members"),
+            "{:?}", json.errors);
+}
+
+
+// Test trying to publish a krate we don't own
+#[test]
+fn publish_not_owned() {
+    let (_b, app, middle) = ::app();
+
+    let mut req = new_req(app.clone(), "foo", "1.0.0");
+    ::mock_user(&mut req, mock_ganksy());
+    ::mock_crate(&mut req, ::krate("foo"));
+
+    let body = r#"{"users":["github:rust-lang:libs"]}"#;
+    ok_resp!(middle.call(req.with_path("/api/v1/crates/foo/owners")
+                           .with_method(Method::Put)
+                           .with_body(body.as_bytes())));
+
+    ::mock_user(&mut req, mock_kitty());
+    let json = bad_resp!(middle.call(req.with_path("/api/v1/crates/new")
+                           .with_body(&new_req_body(::krate("foo"), "2.0.0", vec![]))
+                           .with_method(Method::Put)));
+    assert!(json.errors[0].detail.contains("another user"),
+            "{:?}", json.errors);
+}
+
+// Test trying to publish a krate we do own (but only because of teams)
+#[test]
+fn publish_owned() {
+    let (_b, app, middle) = ::app();
+    let mut req = new_req(app.clone(), "foo", "1.0.0");
+    ::mock_user(&mut req, mock_ganksy());
+    ::mock_crate(&mut req, ::krate("foo"));
+
+    let body = r#"{"users":["github:contain-rs:owners"]}"#;
+    ok_resp!(middle.call(req.with_path("/api/v1/crates/foo/owners")
+                           .with_method(Method::Put)
+                           .with_body(body.as_bytes())));
+
+    ::mock_user(&mut req, mock_kitty());
+    ok_resp!(middle.call(req.with_path("/api/v1/crates/new")
+                           .with_body(&new_req_body(::krate("foo"), "2.0.0", vec![]))
+                           .with_method(Method::Put)));
+}
+
+// Test trying to change owners (when only on an owning team)
+#[test]
+fn change_owners() {
+    let (_b, app, middle) = ::app();
+    let mut req = new_req(app.clone(), "foo", "1.0.0");
+    ::mock_user(&mut req, mock_ganksy());
+    ::mock_crate(&mut req, ::krate("foo"));
+
+    let body = r#"{"users":["github:contain-rs:owners"]}"#;
+    ok_resp!(middle.call(req.with_path("/api/v1/crates/foo/owners")
+                           .with_method(Method::Put)
+                           .with_body(body.as_bytes())));
+
+    ::mock_user(&mut req, mock_kitty());
+    let body = r#"{"users":["FlashCat"]}"#;
+    let json = bad_resp!(middle.call(req.with_path("/api/v1/crates/foo/owners")
+                                       .with_method(Method::Put)
+                                       .with_body(body.as_bytes())));
+    assert!(json.errors[0].detail.contains("don't have permission"),
+            "{:?}", json.errors);
+}
+
```

Comment:
Can you also add tests for:

* Owners can remove a team from a crate
* Team members cannot remove a team from a crate